includes/class-wc-payments-onboarding-service.php

```diff
@@ -1024,4 +1031,66 @@ public function maybe_add_test_drive_settings_to_new_account_request( array $arg
 
 		return $args;
 	}
+
+	/**
+	 * Updates the enabled payment methods for a gateway.
+	 *
+	 * @param WC_Payment_Gateway_WCPay $gateway Payment gateway instance.
+	 * @param array                    $capabilities Provided capabilities.
+	 */
+	public function update_enabled_payment_methods_ids( $gateway, $capabilities = [] ): void {
+		$enabled_gateways = $gateway->get_upe_enabled_payment_method_ids();
+
+		$enabled_payment_methods = array_unique(
+			array_merge(
+				$enabled_gateways,
+				$this->exclude_placeholder_payment_methods( $capabilities )
+			)
+		);
+
+		// Update the gateway option.
+		$gateway->update_option( 'upe_enabled_payment_method_ids', $enabled_payment_methods );
+
+		foreach ( $enabled_payment_methods as $payment_method_id ) {
```

Comment:
Why do we need to also update for individual PMs?

Reply:
That’s a great question. I’m not sure if we should update it for all the payment methods, but I noticed in a few places in the code that synchronization between all gateways is being done. I thought it might be a good idea to follow that approach.

Reply:
I see now that we have overall WooPayments settings (`woocommerce_woocommerce_payments_settings`) probably for `card` and individual settings for each PM that duplicate some of the settings.
![Screenshot 2024-12-20 at 16 51 20](https://github.com/user-attachments/assets/a984ca1a-c138-40a0-82b5-53c9e7ae72d6)

Such a super setup :(

Reply:
@mordeth see previous comment about inline docs.

Reply:
Included an explanation with inline documentation.